documentation/public/user-guide/08-examples/03-jump-to-definition/examples/jump-to-definition.mts

```diff
@@ -0,0 +1,50 @@
+import assert from "node:assert";
+import { TerminalKindExtensions } from "@nomicfoundation/slang/cst";
+import { CompilationUnit } from "@nomicfoundation/slang/compilation";
+import { findTerminalNodeAt } from "../../common/find-terminal-node-at.mjs";
+
+type Target = {
+  file: string;
+  line: number;
+  column: number;
+};
+
+export function jumpToDefinition(unit: CompilationUnit, fileId: string, line: number, column: number): Target | string {
```

Comment:
I wonder if it is more idiomatic to just `throw` in case of errors/invalid input?

```suggestion
export function jumpToDefinition(unit: CompilationUnit, fileId: string, line: number, column: number): Target {
```

Reply:
And we wouldn't need to `assert(typeof result == ...)` in callsites..

Reply:
... same for others

Reply:
It was totally intended 😑